nlp_primitives/number_of_mentions.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Double, NaturalLanguage
+
+from .count_string import CountString
+
+
+class NumberOfMentions(CountString):
+    """Determines the number of mentions in a string.
+
+    Description:
+        Given list of strings, determine the number of mentions
+        in each string. A mention is defined as any word starting
+        with a '@' sign followed by a sequence of alphanumeric characters
+        or underscores.
+
+        If a string is missing, return `NaN`.
+
+    Examples:
+         >>> x = ['@portland@oregon', 'this is a string', '@@@__user1@1and_0#expression']
```

Comment:
My comment from the hashtags also applies here. I think we should consider updating the regex.